### DISCUSSION TITLE
Performance

### DISCUSSION THREAD

**Discussion by leo60228** [fnuecke/oc2#56] [General]
I'm curious as to how well the RISC-V emulation performs.

I'm mainly curious about:
* OC2 Lua vs. CC:Tweaked Lua vs. native Lua
* OC2 "native" vs. native code

I can benchmark this myself later, but I thought it would make sense to start this thread first.

**Comment by fnuecke** [comment]
Seen from user code, Sedna (the RISC-V VM) is definitely slower, not least because it not only emulates Lua but an entire Linux kernel inbetween. I can't quite remember the actual numbers, but it was *very* significant. When running in the mod, Sedna is also throttled to run slower than it potentially could (on my machine, anyway), yet is still fast enough for interactive operation. For the main use-cases I'm interested in right now, that's fast enough :) Still, interested to see what your findings will be!

(I have something I'd like to try speed up the VM a bit, but don't know if that'll work, or when I'll get around to trying it.)

**Reply by atirutw** [reply]
Does this mean OC2 will also use Lua alongside RISC?

**Reply by ToMe25** [reply]
By default [sedna](github.com/fnuecke/sedna) linux has Lua, [micropython](https://micropython.org/)(A python implementation for microcontrollers with limited standard libs) and ash(a unix shell) as available scripting environments.
Meaning while it is not as performant as in OC1, you can still run Lua on a default OC2 computer.

**Reply by atirutw** [reply]
Figured out Lua on OC2 is running on top of Sedna Linux which is on top of RISC-V emu quite a while ago, but thanks.

**Reply by ToMe25** [reply]
I'm sorry, I failed to realize the original comment was almost a year old when answering it.
